snapcraft/project/errors.py

```diff
@@ -58,3 +60,31 @@ def get_brief(self) -> str:
 
     def get_resolution(self) -> str:
         return "This extension may be enabled with the '--enable-experimental-extensions' parameter."
+
+
+class UnsupportedBase(SnapcraftException):
+    def __init__(self, *, base: str) -> None:
+        self.base = base
+
+    def get_brief(self) -> str:
+        return f"Using {self.base!r} as a 'base' or 'build-base' is not supported."
+
+    def get_details(self) -> Optional[str]:
+        if self.base == "core":
+            return (
+                "'core' is currently under Extended Security Maintenance which "
+                "requires a different version of Snapcraft to run."
+            )
+        return None
+
+    def get_resolution(self) -> str:
+        if self.base == "core":
+            return (
+                "Switch to Snapcraft's 4.x channel track or consider upgrading "
+                "to a newer base."
```

Comment:
Perhaps provide command to refresh?

Reply:
I would prefer to enhance that through the documentation link, the procedure is not the same on every OS.